tests/drum/unit/model_metadata/test_model_metadata.py

```diff
@@ -362,6 +363,31 @@ def test_bytes_not_string(self):
         img = np.random.bytes(32)
         assert not DataTypes.is_text(img)
 
+    def test_img_with_nan(self, request):
+        """Test the case where there are also NaN values in along with image values."""
+        yaml_str = input_requirements_yaml(Fields.DATA_TYPES, Conditions.EQUALS, [Values.IMG])
+        schema_dict = self.yaml_str_to_schema_dict(yaml_str)
+        validator = SchemaValidator(schema_dict)
+
+        data = request.getfixturevalue("cats_and_dogs")
+        data = data.drop("class", axis=1)
+        data.loc[np.array(sample(range(len(data)), 10)), :] = np.nan
+        assert validator.validate_inputs(data)
+
+    def test_all_nan_is_num_not_img(self):
```

Comment:
Could try to have more descriptive test case name? I see that you also use the num-validator within the test case:
```suggestion
    def test_img_validator_with_all_nan(self):
```